hax/hax/ha/events.py

```diff
@@ -145,3 +152,78 @@ def ack(self):
            that message using ack method
         """
         self.consumer.ack()
+
+
+class EventProducer():
+    """
+    Module responsible for publishing the event to message bus
+    """
+
+    ADMIN_ID = "hare_admin"
+    __instance = None
+
+    @staticmethod
+    def get_instance():
+        """
+        Static method to fetch the current instance.
+        Performs initialization related to common database and message bus
+        """
+        if not EventProducer.__instance:
+            EventProducer()
+        return EventProducer.__instance
+
+    def __init__(self):
+        logging.debug('Inside EventProducer')
+
+        if EventProducer.__instance is None:
+            EventProducer.__instance = self
+        else:
+            raise Exception("EventProducer is singleton class,"
+                            " use EventProducer.get_instance().")
+
+        message_type = FAULT_TOLERANCE_KEYS.HARE_HA_MESSAGE_TYPE.value
+        self._register_producer(message_type)
+
+        self.producer = MessageProducer(producer_id=HEALTH_EVENT_SOURCES.HARE,
+                                        message_type=message_type)
+
+    def _register_producer(self, message_type,
+                           partitions: int = 1):
+        util: ConsulUtil = ConsulUtil()
+        configpath = util.get_configpath()
+        Conf.load('cortx_conf', configpath)
+
+        message_server_endpoints = Conf.get('cortx_conf',
+                                            'cortx>external>kafka>endpoints')
+        MessageBus.init(message_server_endpoints)
+
+        admin = MessageBusAdmin(admin_id=self.ADMIN_ID)
+        try:
+            if message_type not in admin.list_message_types():
+                admin.register_message_type(message_types=[message_type],
+                                            partitions=partitions)
+        except Exception as e:
+            if "TOPIC_ALREADY_EXISTS" not in str(e):
+                raise(e)
+
+    def create_node_event(self, res_name, health_status, generation_id):
+        self.payload = {
```

Comment:
How the header will be populated?

Reply:
header is automatically populated by HealthEvent(**self.payload) call
Input event is as follows
> {'source': 'hare', 'cluster_id': 'NOT_DEFINED', 'site_id': 'NOT_DEFINED', 'rack_id': 'NOT_DEFINED', 'storageset_id': 'NOT_DEFINED', 'node_id': 'machine-id', 'resource_type': 'node', 'resource_id': 'machine-id', 'resource_status': 'online', 'specific_info': {}}

Output event will be as follows
> {"header": {"version": "1.0", "timestamp": "1648705027.6794617", "event_id": "1648705027.6794617fd6fc7263dd346348fb26936bde6b3a1"}, "payload": {"source": "hare", "cluster_id": "NOT_DEFINED", "site_id": "NOT_DEFINED", "rack_id": "NOT_DEFINED", "storageset_id": "NOT_DEFINED", "node_id": "machine-id", "resource_type": "node", "resource_id": "machine-id", "resource_status": "online", "specific_info": {"generation_id": "localhost"}}}